sw.js

```diff
@@ -1,41 +1,62 @@
-self.addEventListener('install', function(event) {
+const addResourcesToCache = async (resources) => {
+  const cache = await caches.open("v1");
+  await cache.addAll(resources);
```

Comment:
Don't we want to wait until all resources have been added to the cache, when the service worker is installed? 

This is how it was before:

```javascript
self.addEventListener('install', function(event) {
  event.waitUntil(
    caches.open('v1').then(function(cache) {
      return cache.addAll([
        '/sw-test/',
        '/sw-test/index.html',
        '/sw-test/style.css',
        '/sw-test/app.js',
        '/sw-test/image-list.js',
        '/sw-test/star-wars-logo.jpg',
        '/sw-test/gallery/bountyHunters.jpg',
        '/sw-test/gallery/myLittleVader.jpg',
        '/sw-test/gallery/snowTroopers.jpg'
      ]);
    })
  );
});
```

Maybe I'm missing something...

Reply:
Maybe I am :), but I think:

* `addResourcesToCache` is going to return a promise, anyway, because it's async
* `waitUntil` will wait until that returned promise is settled
* `await cache.addAll(resources)` won't affect that - it would only affect code inside `addResourcesToCache`. But there isn't any, we return right after that line anyway.

Put another way: AIUI, `await` inside `addResourcesToCache` isn't going to affect the code that called `addResourcesToCache` - either way, `addResourcesToCache` will return right away with a promise that will be settled later on, when the async stuff in `addResourcesToCache` has settled.

> Don't we want to wait until all resources have been added to the cache, when the service worker is installed?

What matter for this is that we are calling `waitUntil` on the promise that `addResourcesToCache` returns.

Does that make sense?

Reply:
> await cache.addAll(resources) won't affect that - it would only affect code inside addResourcesToCache. But there isn't any, we return right after that line anyway.

When calling (awaiting) this version of addResourcesToCache, the promise from addResourcesToCache is going to be resolved when cache.addAll(resources) is still in pending state.

```js
const addResourcesToCache = async (resources) => {
  const cache = await caches.open("v1");
  cache.addAll(resources);
}
```

For example, this

```js
const cache = {
  async addAll(resources) {
    await new Promise((resolve) => setTimeout(resolve, 0))
    console.log('resources have been added to cache')
  },
}

const addResourcesToCache = async (resources) => {
  cache.addAll(resources)
}

const main = async () => {
  await addResourcesToCache([])
  console.log('addResourcesToCache has finished')
}

main()

// prints out:
// addResourcesToCache has finished
// resources have been added to cache
```


Another problem would be error-handling with try-catch:


```js
const cache = {
  async addAll(resources) {
    await new Promise((resolve) => setTimeout(resolve, 0))
    throw new Error('oops')
  },
}

const addResourcesToCache = async (resources) => {
  cache.addAll(resources)
}

const main = async () => {
  try {
    await addResourcesToCache([])
  } catch (error) {
    console.log('error handling here')
  }
}

main()


// prints:
// Uncaught (in promise) Error: oops
```

Thoughts? :)

Reply:
OK I am convinced!